rust/private/rustc.bzl

```
@@ -780,7 +780,7 @@ def construct_arguments(
             rustc_flags.add("--codegen=linker=" + ld)
             rustc_flags.add_joined("--codegen", link_args, join_with = " ", format_joined = "link-args=%s")
 
-        _add_native_link_flags(rustc_flags, dep_info, linkstamp_outs, ambiguous_libs, crate_info.type, toolchain, cc_toolchain, feature_configuration)
+        _add_native_link_flags(rustc_flags, dep_info, linkstamp_outs, ambiguous_libs, crate_info.type, toolchain, cc_toolchain, feature_configuration, ctx.attr.experimental_use_whole_archive_for_native_deps)
```

```suggestion
        _add_native_link_flags(rustc_flags, dep_info, linkstamp_outs, ambiguous_libs, crate_info.type, toolchain, cc_toolchain, feature_configuration, attr.experimental_use_whole_archive_for_native_deps)
```

We don't want to access attributes directly from `ctx` because that causes failures when we reach the code via an aspect -> in such cases one needs `ctx.rule.attr` instead of `ctx.attr`.

---

Thanks! Good to know!